### PR TITLE
migrate to debian:stretch-slim

### PR DESCRIPTION
**Pull request by Lorac**
- Can use Qt5 (5.7.1) from debian stretch
- Keep Eigen3.2.5 for now
- Need to use GCC 4.9.2 until we migrate to gcc 6.2
- Patch ccache because we don't want it to build the documentation

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,46 +1,81 @@
...
-							build-essential \
-							software-properties-common 
+							patch \
+							qt5-default \
+							qtbase5-dev-tools \
+							qtdeclarative5-dev \
+							wget
 
+RUN echo 'deb http://httpredir.debian.org/debian/ jessie main contrib non-free' >> /etc/apt/sources.list
+RUN echo 'deb-src http://httpredir.debian.org/debian/ jessie main contrib non-free' >> /etc/apt/sources.list
```
Comment (jeff-dagenais) [inline Dockerfile]: No need for deb-src in a docker image. This is to get package's source code using `apt-get source packageX`

File: Dockerfile [hunk 2]
```diff
@@ -1,46 +1,81 @@
...
-RUN add-apt-repository ppa:beineri/opt-qt57-xenial \
-	&& apt-get update \
+RUN apt-get update \
 	&& apt-get install -y --no-install-recommends --no-install-suggests \
-							qt57declarative \
-							libgl1-mesa-dev \
-	&& apt-get purge -y software-properties-common --auto-remove
-	
+							gcc=4:4.9.2-2 \
+							g++=4:4.9.2-2 \
```
Comment (jeff-dagenais) [inline Dockerfile]: remove extra gcc package

File: Dockerfile [hunk 3]
```diff
@@ -1,46 +1,79 @@
...
-RUN add-apt-repository ppa:beineri/opt-qt57-xenial \
-	&& apt-get update \
+RUN apt-get update \
 	&& apt-get install -y --no-install-recommends --no-install-suggests \
-							qt57declarative \
-							libgl1-mesa-dev \
-	&& apt-get purge -y software-properties-common --auto-remove
-	
+							g++=4:4.9.2-2 \
+	&& head -n -1 /etc/apt/sources.list
```
Comment (jeff-dagenais) [inline Dockerfile]: this head line does nothing but print to standard out during docker build

Comment (Lorac) [inline Dockerfile]: removed

**Pull request merged**